### ISSUE TITLE
Conversations does not sync logs after about 2 weeks

### ISSUE THREAD

**Issue by allo-** [iNPUTmice/Conversations#2830]
#### General information

* **Version:**1.23.6
* **Device:** Galaxy Note 10.1
* **Android Version:** Android 4.4 (Lineage OS)
* **Server name:** self hosted
* **Server software:** prosody 0.9.12
* **Installed server modules:** MAM
* **Conversations source:** PlayStore

#### Steps to reproduce

1.Use two devices with conversations
2. Have one device be offline for about two weeks
3. Have a server configured to keep MAM for 1 year ``archive_expires_after = "1y"``

#### Expected result
Conversation syncs logs when the device is no longer offline than 1 year

#### Actual result

After some time (i.e. about two weeks) conversations does not sync old logs anymore.

**Comment by licaon-kter** [comment]
Encrypted OMEMO chats? Unencrypted?

Can you try with Prosody 0.10?

**Comment by allo-** [comment]
OMEMO encrypted. cannot easily try with newer prosidy, but the 0.9 has installed the all access Plugin for omemo.

**Comment by allo-** [comment]
Do you have an idea what may be the issue? It's hard to debug when I only see the isdue when I usr a seldom used device. I hoped you may have an idea where to search.

**Comment by licaon-kter** [comment]
There are no logs?

The user has "Archive:Always" in Conversations?

**Comment by allo-** [comment]
Log synchronization works after the device was used again for some time. Current experience is, that about two weeks stops the synchronization and after that ie. 12 hours are loaded again.
Example:

- device started, no logs synchronized at all
- some chatting from another device, sync works nicely.
- device turned off
- some chatting on an other device
- device turned on after about 12 hours
- logs are synced again.

I do not know exactly how long the device can be offline before it stops synchronizing.

Are the other things with timeout than the MAM archive? Are OMEMO keys expiring after some (shorter) time?

**Comment by iNPUTmice** [comment]
There is a hard coded limit to 14 days.

Closing gaps is extremly difficult so we always have to download the entire history. However that's not very good either. So two weeks is the compromise. Assuming that if you haven't used the device in two weeks it's probably not that important.

**Comment by allo-** [comment]
I don't think thus is reasonable for all use cases, especially since I do not get the first 14 days after the start of the break or the last 14 before the end synced. Can we get an expert option for this?

**Comment by 9662** [comment]
What about a 14 day initial fetch and then a "fetch older" action? Something like a menu item, gesture, whatever.

**Comment by 9662** [comment]
Fetching older stuff is important sometimes, e.g., to go back to a link sent in a chat or a file, etc.

**Comment by kousu** [comment]
Is the hardcoded limit still in place? I just scrolled back to September 12th in one conversation -- and as today is October 7th, that's more than 2 weeks ago. And it seems I can keep scrolling back further and further.

I thought there was a MAM bug in my server, but it turned out I just had Settings > Expert settings > Automatic message deletion = 1 week.  Then I went debugging, found this thread, figured out my mistake, set it to 6 months, and now I seem to have indefinite (well, presumably 6 month) scrollback.

**General information**

    **Version:** 2.5.5+fcr
    Device: Moto E
    Android Version: Lineage OS 14.1 (Android 7.1.2)
    Server name: self hosted
    Server software: prosody 0.10.0
    Installed server modules: mod_mam
    Conversations source: F-Droid
